src/OrchardCore.Modules/OrchardCore.Facebook/Services/IFacebookService.cs

```diff
@@ -1,16 +0,0 @@
-using System.Collections.Generic;
-using System.ComponentModel.DataAnnotations;
-using System.Threading.Tasks;
-using OrchardCore.Facebook.Settings;
-
-namespace OrchardCore.Facebook.Services
-{
-    public interface IFacebookService
```

Comment:
I'm OK with the `OAuthSettingsService<TAuthenticationSettings>` base class, but why is it good to remove the service interfaces? We lose the ability to override them (for extensibility and testing).

Reply:
> We lose the ability to override them (for extensibility and testing).

Totally agree, I just react toSeb feedback on the meeting, but I think it's fine to leave it as it is

Reply:
@sebastienros please elaborate, then.

Reply:
May be because each service has default implementation, but still I'm agree with you :)

Reply:
May be because each service has default implementation, but still I'm agree with you :)

Reply:
I still don't think we should remove the service interfaces. It's a step back.

Reply:
I'm not sure why we need if we have an abstract class?!!

Reply:
The same reason as we do it for services generally: loose coupling. However, these classes aren't even abstract. All are concrete types that are injected, removing the option to override them.

Reply:
I didn't get you, I'm not sure why we should keep interfaces with the same methods while we can introduce a new abstract class to do the same thing

Reply:
I will bring the interfaces again

Reply:
Zoltan while we have many interfaces, I think we need only one for all services that could be implemented in the abstract class, right?

Reply:
Not really. If I want to override `FacebookService`, I need a specific interface for that, i.e. `IFacebookService` as it is in `main`. A common interface won't work for this.

Reply:
Could you please let me know how all the interfaces will interact with the newly added abstract class?

Reply:
They don't need to, just e.g. `FacebookService` needs to implement `IFacebookService`, as it was. All these interfaces can remain as they were, and since the abstract class implements them, in e.g. `FacebookService` you don't need to add anything else.

Reply:
That means nothing to be removed except a few types have been added

Reply:
Well, up to you how to go forward. The interfaces should stay because they have a use, they can't just be removed, as it will break people's code.

Reply:
Return all interfaces will make things harder because the abstract class should implement all the auth interfaces which do not make sense, all are equivalent except the settings type. If we introduce a generic type we will find ourselves using a single interface instead which is why I moved the implementation into an abstract class

Reply:
I can only repeat myself. Refactoring that breaks all consumers of these authentication services is not suitable. We have more than 400 interfaces in OC for such services, for a good reason. I really don't want to debate further why we should keep them. By looking at the code it seems to me that the only thing you need to do is re-add the interfaces and revert the dependency registrations, nothing else to change. It's something really simple.

Without the interfaces, I don't approve this PR. It's your call whether you want to rework the PR to keep them, or close it.

Reply:
As I said before my PR is based on @sebastienros suggestion nothing else. Moreover, I never mind reverting all the removed interfaces but the thing that I still do not get is how the interfaces will be used in the introduced abstract class

If you think this will take so long, please close the PR, I don't want to continue in something I wasn't convinced with

Thanks

Reply:
I suggest you just re-add the interfaces to see in action that nothing else with the base class needs to change.

Reply:
I just reverted the interfaces back as you said but again a generic interface make sense to me rather than have 4 - 6 interfaces with the same methods